examples/timeseries_peak_finding.py

```diff
@@ -0,0 +1,78 @@
+"""
+==============================
+Find Peaks in SunPy TimeSeries
+==============================
+
+This example illustrates how to find minimum or maximum peaks in a TimeSeries.
+"""
+
+##############################################################################
+# Start by importing the necessary modules.
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+import sunpy.data.sample
+from sunpy.timeseries import TimeSeries
+
+##############################################################################
+# We will now create a TimeSeries object from an observational data source and 
+# do analysis on its truncated version.
+
+ts_noaa_ind = sunpy.timeseries.TimeSeries(
+    sunpy.data.sample.NOAAINDICES_TIMESERIES, source='NOAAIndices')
+my_timeseries = ts_noaa_ind.truncate(0,100)
+my_timeseries.peek()
+
+##############################################################################
+# Now we take the column 'sunspot SWO' of this TimeSeries and try to find it's
+# peaks. For peak finding, we first decide a DELTA value which controls how
+# much difference between values in the TimeSeries defines a peak. Then we
+# iterate over the data values and consider a point to be a maximum peak if it
+# has the maximal value, and was preceded (to the left) by a value lower by 
+# DELTA. Similar logic applies to find the minimum valued peaks.
+
+series = my_timeseries.data['sunspot SWO']
+DELTA = 10
+# Set inital values
+mn, mx = np.Inf, -np.Inf
+minpos, maxpos = np.NaN, np.NaN
+minpeaks = []
+maxpeaks = []
+lookformax = True
+# Iterate over data values
+for i in np.arange(len(series)):
+    this = series[i]
+    if this > mx:
+        mx = this
+        mxpos = series.index[i]
+    if this < mn:
+        mn = this
+        mnpos = series.index[i]
+    if lookformax:
+        if this < mx-DELTA:
+        	# a local maxima peak
+            maxpeaks.append((mxpos, mx))
+            mn = this
+            mnpos = series.index[i]
+            lookformax = False
+    else:
+        if this > mn+DELTA:
+        	# a local minima peak
+            minpeaks.append((mnpos, mn))
+            mx = this
+            mxpos = series.index[i]
+            lookformax = True
```

Comment:
I am not sure how I feel about the way this is laid out. But right now I don't have a better solution off my head.

Reply:
As this is it will fail to find min/max peaks if they happen at the beginning/end of the series respectively. We could discuss whether a peak is such when there's only data on one side of it, but in the data you are using the first max is shown, but not the last. If the first one is considered a peak then the last one should be too.

Reply:
I think since we are iterating sequentially from left to right, the current algorithm will always consider first value to be peak but not necessarily the last, as it always compares to the left. Should we do something about the first value then?